src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -818,7 +818,7 @@ def forward(
         if inputs_embeds is None:
             inputs_embeds = self.wte(input_ids)
         position_embeds = self.wpe(position_ids)
-        hidden_states = inputs_embeds + position_embeds
+        hidden_states = inputs_embeds + position_embeds.to(inputs_embeds.device)
```

Comment:
FYI reviewers, this fix was taken from here:
https://github.com/huggingface/transformers/blob/5fa35344755d8d9c29610b57d175efd03776ae9e/src/transformers/models/whisper/modeling_whisper.py#L1270

See https://github.com/huggingface/transformers/pull/30836#discussion_r1604774353 for associated discussion.